gc.c

```diff
@@ -2817,12 +2810,6 @@ obj_free(rb_objspace_t *objspace, VALUE obj)
       case T_MOVED:
 	break;
       case T_ICLASS:
-	/* Basically , T_ICLASS shares table with the module */
-        if (FL_TEST(obj, RICLASS_IS_ORIGIN) &&
-                !FL_TEST(obj, RICLASS_ORIGIN_SHARED_MTBL)) {
-            /* Method table is not shared for origin iclasses of classes */
-            rb_id_table_free(RCLASS_M_TBL(obj));
-        }
 	if (RCLASS_CALLABLE_M_TBL(obj) != NULL) {
 	    rb_id_table_free(RCLASS_CALLABLE_M_TBL(obj));
 	}
```

Comment:
This method table is managed by the GC now, right?  Can we remove this if statement?

Reply:
`CALLABLE_M_TBL` is still manually managed. It's not shared between objects AFAIK. I'm not sure if there would be any benefits from moving this table onto the GC heap.